crates/bevy_reflect/src/list.rs

```diff
@@ -401,7 +401,7 @@ impl<'a> Iterator for ListIter<'a> {
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
         let value = self.list.get(self.index);
-        self.index += 1;
+        self.index += value.is_some() as usize;
```

Comment:
Non-blocker: While we're doing this, it would be great if we updated the other iterator types in the crate like `MapIter`, `TupleFieldIter`, etc.

Reply:
I don't mind doing them now. Is it correct to update the next function in these:
	modified:   src/array.rs
	modified:   src/enums/enum_trait.rs
	modified:   src/map.rs
	modified:   src/struct_trait.rs
	modified:   src/tuple.rs
	modified:   src/tuple_struct.rs
but not in:
 src/path/parse.rs

Reply:
Yep! 😄